geode-core/src/distributedTest/java/org/apache/geode/distributed/internal/tcpserver/TcpServerBackwardCompatDUnitTest.java

```diff
@@ -180,40 +150,43 @@ public void run2() throws CacheException {
     });
 
     // Restart first locator in new VM.
-    // props.setProperty(DistributionConfig.START_LOCATOR_NAME, host.getHostName() + "["+port0+"]");
     locatorRestart0
         .invoke(new CacheSerializableRunnable("Restarting first locator on port " + port0) {
 
           @Override
           public void run2() throws CacheException {
-            try {
-              TcpServer.TESTVERSION -= 100;
-              TcpServer.OLDTESTVERSION -= 100;
-              TcpServer.getGossipVersionMapForTestOnly().put(TcpServer.TESTVERSION,
-                  Version.CURRENT_ORDINAL);
-              TcpServer.getGossipVersionMapForTestOnly().put(TcpServer.OLDTESTVERSION,
-                  Version.GFE_57.ordinal());
-              // assertIndexDetailsEquals("Gossip Version and Test version are not same",
-              // TcpServer.GOSSIPVERSION, TcpServer.TESTVERSION);
-              // assertIndexDetailsEquals("Previous Gossip Version and Test version are not same",
-              // TcpServer.OLDGOSSIPVERSION, TcpServer.OLDTESTVERSION);
-
-              Locator.startLocatorAndDS(port0, logFile0, props);
-
-              // Start a gossip client to connect to first locator "locator0".
-              FindCoordinatorRequest req = new FindCoordinatorRequest(
-                  new InternalDistributedMember(SocketCreator.getLocalHost(), 1234));
-              FindCoordinatorResponse response = null;
-
-              response = (FindCoordinatorResponse) new TcpClient()
-                  .requestToServer(SocketCreator.getLocalHost(), port0, req, 5000);
-              assertNotNull(response);
-
-            } catch (Exception e) {
-              org.apache.geode.test.dunit.Assert.fail(
-                  "Locator0 start failed with Gossip Version: " + TcpServer.GOSSIPVERSION + "!", e);
-            }
+            restartLocator(port0, logFile0, props);
           }
         });
   }
+
+  protected void restartLocator(int port0, File logFile0, Properties props) {
+    try {
+      TcpServer.TESTVERSION -= 100;
+      TcpServer.OLDTESTVERSION -= 100;
+      TcpServer.getGossipVersionMapForTestOnly().put(TcpServer.TESTVERSION,
+          Version.CURRENT_ORDINAL);
+      TcpServer.getGossipVersionMapForTestOnly().put(TcpServer.OLDTESTVERSION,
+          Version.GFE_57.ordinal());
+
+
+      Locator.startLocatorAndDS(port0, logFile0, props);
+
+      // Start a gossip client to connect to first locator "locator0".
+      FindCoordinatorRequest req = new FindCoordinatorRequest(
+          new InternalDistributedMember(SocketCreator.getLocalHost(), 1234));
+      FindCoordinatorResponse response = null;
+
+      response = (FindCoordinatorResponse) new TcpClient()
+          .requestToServer(SocketCreator.getLocalHost(), port0, req, 5000);
+      assertNotNull(response);
+
+    } catch (LocatorCancelException e) {
```

Comment:
Ignore if you want: If you change the caller of restartLocator to be a lambda call, then you can delete these catch blocks and change the restartLocator method to have "throws Exception":
```
locatorRestart0.invoke(() -> restartLocator(port0, logFile0, props));

protected void restartLocator(int port0, File logFile0, Properties props) throws Exception {
```

Reply:
Done. Cleaned up this whole file...